server/src/main/java/org/opensearch/plugins/PluginInfo.java

```diff
@@ -214,10 +262,45 @@ public static PluginInfo readFromProperties(final Path path) throws IOException
         }
 
         final String opensearchVersionString = propsMap.remove("opensearch.version");
-        if (opensearchVersionString == null) {
-            throw new IllegalArgumentException("property [opensearch.version] is missing for plugin [" + name + "]");
+        final String dependenciesValue = propsMap.remove("dependencies");
+        if (opensearchVersionString == null && dependenciesValue == null) {
+            throw new IllegalArgumentException(
+                "Either [opensearch.version] or [dependencies] property must be specified for the plugin [" + name + "]"
+            );
         }
-        final Version opensearchVersion = Version.fromString(opensearchVersionString);
+        if (opensearchVersionString != null && dependenciesValue != null) {
+            throw new IllegalArgumentException(
+                "Only one of [opensearch.version] or [dependencies] property can be specified for the plugin [" + name + "]"
+            );
+        }
+
+        final List<SemverRange> opensearchVersionRanges = new ArrayList<>();
+        if (opensearchVersionString != null) {
+            opensearchVersionRanges.add(SemverRange.fromString(opensearchVersionString));
+        } else {
+            ObjectMapper mapper = new ObjectMapper().configure(
```

Comment:
You don't need to bring whole `jackson-databind` here, the `jackson-core` + XContent parsing we already have (as we discussed) is fully sufficient:

```
           // This could be made a static / final instance, no need to create every time
            final JsonFactory jsonFactory = new JsonFactory();
            jsonFactory.configure(JsonReadFeature.ALLOW_UNQUOTED_FIELD_NAMES.mappedFeature(), true);
            
            Map<String, Object> dependenciesMap = null;
            try (final JsonXContentParser parser = new JsonXContentParser(NamedXContentRegistry.EMPTY, DeprecationHandler.IGNORE_DEPRECATIONS, jsonFactory.createParser(dependenciesValue))) {
                dependenciesMap = parser.map(); 
            }
```

Reply:
Updated based on recommendation.